packages/plugin-essentials/sources/suggestUtils.ts

```diff
@@ -335,3 +338,17 @@ export async function fetchDescriptorFrom(ident: Ident, range: string, {project,
   return structUtils.makeDescriptor(bestLocator, structUtils.makeRange({protocol, source, params, selector}));
 }
 
+export async function makeTemporaryCache() {
+  return await xfs.mktempPromise(async cacheDir => {
+    const configuration = Configuration.create(cacheDir);
+
+    configuration.useWithSource(cacheDir, {
+      // Don't pollute the mirror with temporary archives
+      enableMirror: false,
+      // Don't spend time compressing what gets deleted later
+      compressionLevel: 0,
+    }, cacheDir, {overwrite: true});
+
+    return new Cache(cacheDir, {configuration, check: false, immutable: false});
```

Comment:
The temp folder will be deleted on this return, you'll either need to detach it or avoid passing a function.

Reply:
Thanks, fixed!